x-pack/plugins/observability_solution/infra/public/components/shared/templates/infra_page_template.tsx

```diff
@@ -118,10 +130,22 @@ export const InfraPageTemplate = ({
     />;
   }
 
+  const evaluateNoDataConfig = () => {
+    if (entitySummaryStatus === 'failure') {
+      return noDataConfig;
+    }
+
+    if (entitySummaryStatus !== 'success') {
```

Comment:
Anything that is not **failure** and not **success** is **loading or not_initiated**, shouldn't we show a loading spinner in this case? Otherwise, the page will first blink with some empty components and then  display the no data screen.

Reply:
We didn’t have a proper loading state before (as shown in the attached GIFs), even when the `/hasData` request was in progress.

When the `noDataConfig` is loading (waiting for the `/hasData`), it returns as `undefined`. 

<img width="536" alt="Screenshot 2024-10-16 at 11 39 48" src="https://github.com/user-attachments/assets/6ce65bf3-9942-4ae1-a99c-562eff162394">

To avoid disrupting the base logic, I maintained the same approach and have `undefined` as the `noDataConfig` until everything is fully loaded.